sunbird-rc-esignet-integration-impl/src/test/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCAuthenticaionServiceTest.java

```diff
@@ -0,0 +1,300 @@
+package io.mosip.esignet.sunbirdrc.integration.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.mosip.esignet.api.dto.*;
+import io.mosip.esignet.api.exception.KycAuthException;
+import io.mosip.esignet.api.exception.KycExchangeException;
+import io.mosip.esignet.api.exception.SendOtpException;
+import io.mosip.esignet.api.util.ErrorConstants;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(MockitoJUnitRunner.class)
+@Slf4j
+public class SunbirdRCAuthenticaionServiceTest {
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    private ObjectMapper objectMapper;
+
+    @InjectMocks
+    private SunbirdRCAuthenticationService sunbirdRCAuthenticationService;
+
+
+
+    @Test
+    public void initializeWithValidConfig_thenPass() throws KycAuthException {
+
+        List<Map<String,String>> fieldDetailList = List.of(Map.of("id","policyNumber","type","string","format","string"));
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "fieldDetailList", fieldDetailList);
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "idField", "policyNumber");
+        ReflectionTestUtils.invokeMethod(sunbirdRCAuthenticationService, "initialize");
+
+    }
+
+    @Test
+    public void initializeWithInValidConfig_thenFail() {
+
+        Throwable thrownException = Assert.assertThrows(Throwable.class,
+                () -> ReflectionTestUtils.invokeMethod(sunbirdRCAuthenticationService, "initialize"));
+
+
+        Assert.assertTrue(thrownException instanceof UndeclaredThrowableException);
+        Throwable actualException = ((UndeclaredThrowableException) thrownException).getUndeclaredThrowable();
+        Assert.assertTrue(actualException instanceof KycAuthException);
+        Assert.assertEquals("sunbird-rc authenticator field is not configured properly", actualException.getMessage());
```

Comment:
Error messages can have minor changes over period of time. Can we compare using an error code here ?

Reply:
Also mentioned here. https://github.com/mosip/digital-credential-plugins/pull/12#discussion_r1455969193

Reply:
we dont use any error code for this exeception in service class only error message